subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuitesIntegrationTest.groovy

```diff
@@ -784,4 +784,94 @@ class TestSuitesIntegrationTest extends AbstractIntegrationSpec {
         fails("integrationTest")
         failure.assertHasErrorOutput("Configuring tests failed")
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/19065")
+    def "test suites can add platforms in convention plugins using #format"() {
```

Comment:
LGTM but I wonder if this test is a little specialized? Did we determine the root cause why this only happened when adding the platform via a plugin? I guess I'm curious if this is a test-suites thing, or if we can also reproduce this without test suites by having one project depend on a platform via its main source set as opposed to its test

Reply:
Without involving test suites, the `platform` method calls directly to to `DefaultDependencyHandler#platform(Object)` to create the dependency, which results in a call to `DefaultDependencyFactory#createDependency(Object)`.  With test suites, the call invokes `DefaultDependencyFactory#create(CharSequence)` to create a dependency.  These uses different paths because of the strongly typed `dependencies` block used by Test Suites, so this issue would only appear when suites are involved.

What I was wondering looking at this was why `createDependency(Object)`'s `else` branch didn' forward to `create(CharSequence)`, but I figured there's no way to funnel every dependency creation down to single private method anyways, so I'd just leave it.

You're right about the test being too specialized, though.  I had originally thought the convention plugin was part of the issue given the original reproducer, but it's not involved.  I'll remove that and simplify the test.

Reply:
Makes sense, thanks for the explanation 

> What I was wondering looking at this was why `createDependency(Object)`'s `else` branch didn' forward to `create(CharSequence)`, but I figured there's no way to funnel every dependency creation down to single private method anyways, so I'd just leave it.

There's definitely some weird looking stuff in this area, I wouldn't be surprised if some of it couldn't be cleaned up